### ISSUE TITLE
vsgexecutecommands crashes on windows

### ISSUE THREAD

**Issue by sbrkopac** [vsg-dev/vsgExamples#285]
Call Stack:

```
 	vsgexecutecommands.exe!vsg::RenderPass::operator unsigned __int64() Line 95	C++
 	vsgexecutecommands.exe!vsg::GraphicsPipeline::Implementation::Implementation(vsg::Context & context, vsg::Device * device, const vsg::RenderPass * renderPass, const vsg::PipelineLayout * pipelineLayout, const std::vector<vsg::ref_ptr<vsg::ShaderStage>,std::allocator<vsg::ref_ptr<vsg::ShaderStage>>> & shaderStages, const std::vector<vsg::ref_ptr<vsg::GraphicsPipelineState>,std::allocator<vsg::ref_ptr<vsg::GraphicsPipelineState>>> & pipelineStates, unsigned int subpass) Line 197	C++
 	vsgexecutecommands.exe!vsg::Inherit<vsg::Object,vsg::GraphicsPipeline::Implementation>::create<vsg::Context &,vsg::ref_ptr<vsg::Device> &,vsg::ref_ptr<vsg::RenderPass> &,vsg::ref_ptr<vsg::PipelineLayout> &,std::vector<vsg::ref_ptr<vsg::ShaderStage>,std::allocator<vsg::ref_ptr<vsg::ShaderStage>>> &,std::vector<vsg::ref_ptr<vsg::GraphicsPipelineState>,std::allocator<vsg::ref_ptr<vsg::GraphicsPipelineState>>> &,unsigned int &>(vsg::Context & <args_0>, vsg::ref_ptr<vsg::Device> & <args_1>, vsg::ref_ptr<vsg::RenderPass> & <args_2>, vsg::ref_ptr<vsg::PipelineLayout> & <args_3>, std::vector<vsg::ref_ptr<vsg::ShaderStage>,std::allocator<vsg::ref_ptr<vsg::ShaderStage>>> & <args_4>, std::vector<vsg::ref_ptr<vsg::GraphicsPipelineState>,std::allocator<vsg::ref_ptr<vsg::GraphicsPipelineState>>> & <args_5>, unsigned int & <args_6>) Line 37	C++
 	vsgexecutecommands.exe!vsg::GraphicsPipeline::compile(vsg::Context & context) Line 183	C++
 	vsgexecutecommands.exe!vsg::BindGraphicsPipeline::compile(vsg::Context & context) Line 285	C++
 	vsgexecutecommands.exe!vsg::StateGroup::compile(vsg::Context & context) Line 59	C++
 	vsgexecutecommands.exe!vsg::CompileTraversal::apply(vsg::StateGroup & stateGroup) Line 259	C++
 	vsgexecutecommands.exe!vsg::Inherit<vsg::Group,vsg::StateGroup>::accept(vsg::Visitor & visitor) Line 70	C++
 	vsgexecutecommands.exe!vsg::Group::t_traverse<vsg::Group,vsg::Visitor>(vsg::Group & node, vsg::Visitor & visitor) Line 37	C++
 	vsgexecutecommands.exe!vsg::Group::traverse(vsg::Visitor & visitor) Line 40	C++
 	vsgexecutecommands.exe!vsg::CompileTraversal::apply(vsg::View & view) Line 353	C++
 	vsgexecutecommands.exe!vsg::View::t_accept<vsg::View,vsg::Visitor>(vsg::View & node, vsg::Visitor & visitor) Line 54	C++
 	vsgexecutecommands.exe!vsg::View::accept(vsg::Visitor & visitor) Line 59	C++
 	vsgexecutecommands.exe!vsg::Group::t_traverse<vsg::Group,vsg::Visitor>(vsg::Group & node, vsg::Visitor & visitor) Line 37	C++
 	vsgexecutecommands.exe!vsg::Group::traverse(vsg::Visitor & visitor) Line 40	C++
 	vsgexecutecommands.exe!vsg::CompileTraversal::apply(vsg::CommandGraph & commandGraph) Line 287	C++
 	vsgexecutecommands.exe!vsg::Inherit<vsg::CommandGraph,vsg::SecondaryCommandGraph>::accept(vsg::Visitor & visitor) Line 70	C++
 	vsgexecutecommands.exe!vsg::Viewer::compile(vsg::ref_ptr<vsg::ResourceHints> hints) Line 365	C++
>	vsgexecutecommands.exe!main(int argc, char * * argv) Line 280	C++
```
Looks like on line 197 of GraphicsPipeline renderPass is nullptr:

```
pipelineInfo.renderPass = *renderPass;
```

**Comment by robertosfield** [comment]
I just tried vsgexecutecommands on this Ubuntu 22.04 laptop and it also seg faults.

Would be worth trying older versions to see when this regression happened.  I can't do this right away.

**Comment by robertosfield** [comment]
I have used git bisect to figure out which commit caused the regression in vsgexecutecommands and it looks like this one from November 14th caused the regression:

https://github.com/vsg-dev/VulkanSceneGraph/commit/c89774cb

**Comment by robertosfield** [comment]
I have fixed the crash so that vsgexecutecommands works once more. The fix is checked into VSG master and is more extensive than just a bug fix:

https://github.com/vsg-dev/VulkanSceneGraph/commit/b8a027c7d4361de8557816e7229a99c8c58e44a1

**Comment by sbrkopac** [comment]
It renders now but closing window 1 causes window 2 to hang and I have to force close it.

**Comment by robertosfield** [comment]
I just tried closing window 1 then 2, then closing 2 then 1 and in both cases the application closes both windows and exits without error.  This is on Kubuntu 22.04.

What platform are you using?

**Comment by sbrkopac** [comment]
Windows 10 - MSVC 2022. I'll dig a little bit when I get some time.

**Comment by robertosfield** [comment]
I will try on my Windows 11 system tomorrow.

**Comment by robertosfield** [comment]
I have tested under Windows 11 and found that vsgexecutecommands exits correctly when pressing escape or closing the second window first via the close button, but hangs if I close the first window.

I have also tried vsgwindows but it runs fine and it doesn't matter which order of closing.

I don't have ideas to the cause of this yet, it could be some execute commands dependency.  I have some other larges tasks on right now so can't investigate further right away.
